lib/payment_processor/clients/braintree/transaction.rb

```diff
@@ -62,14 +53,14 @@ def options
               last_name: @user[:last_name],
               email: @user[:email]
             }
-          }.tap{ |opts| opts[:customer_id] = customer.customer_id if customer }
+          }.tap{ |opts| opts[:customer_id] = @customer.customer_id if @customer }
         end
 
         # Don't store payment method in Braintree's vault if the
         # customer already exists.
         #
         def store_in_vault?
-          customer.nil?
+          @customer.nil?
```

Comment:
The customer is created in `Payment`: https://github.com/SumOfUs/Champaign/blob/nested/app/models/payment.rb#L63